python/python/lance/dataset.py

```diff
@@ -418,7 +418,7 @@ def sample(
         table : Table
         """
         total_num_rows = self.count_rows()
-        indices = random.sample(range(total_num_rows), num_rows)
+        indices = sorted(random.sample(range(total_num_rows), num_rows))
```

Comment:
add explanation on the benefits of indices being sorted

Reply:
This is a breaking API change. Could we add an argument `sorted: bool=False` to this function so users can optionally enable this?

Reply:
`sorted` indicates to sort by value. Shall we use `sequential` or sth ?

Reply:
Perhaps `randomize_order: bool=True`? Then if `randomize_order` is false we call `sorted`. What do you think of that?

Reply:
Ok, it sounds better.